Sources/Build/llbuild.swift

```diff
@@ -63,14 +63,17 @@ public struct LLBuildManifestGenerator {
             newTarget.cmds.insert(phonyCommand)
             otherTargets.append(newTarget)
 
-            if !isTest {
-                main.outputs += newTarget.outputs
-                main.cmds += newTarget.cmds
+            if buildByDefault {
+                if !isTest {
+                    main.outputs += newTarget.outputs
+                    main.cmds += newTarget.cmds
+                }
+
+                // Always build everything for the test target.
+                test.outputs += newTarget.outputs
```

Comment:
Does this mean we will end up compiling tests from other packages for when building tests?

Reply:
No, because the code only runs when`buildByDefault` is `true`, which is set to `plan.graph.reachableTargets.contains(target)` on line 91. See below the [vapor](https://github.com/vapor/vapor) build manifest output from a master build of swiftpm vs with this branch:

[llbuild_manifests.zip](https://github.com/apple/swift-package-manager/files/1511983/llbuild_manifests.zip)